specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,186 @@
+# Background
+We have heard asks for a WebView2 API to easily track the `WebView2 Runtime`'s
+browser process exit. Manually waiting for the process to exit requires
+additional work on the host app, so we are proposing the `BrowserProcessExited`
+event. The `ProcessFailed` already lets app developers handle unexpected browser
+process exits for a WebView, this new API lets you listen to both expected and
+unexpected process termination from the `ICoreWebView2Environment3` interface so
+you can, e.g., cleanup the user data folder when it's no longer in use. In this
+document we describe the new API. We'd appreciate your feedback.
+
+
+# Description
+The `BrowserProcessExited` event allows developers to subscribe event handlers
+to be run when the `WebView2 Runtime`'s browser process associated to a
+`CoreWebView2Environment` terminates. A key scenario is cleanup of the use data
+folder used by the `WebView2 Runtime`, which is locked while the runtime's
+browser process is active.
+
+This event is raised for both expected and unexpected browser process
+termination. The `ICoreWebView2BrowserProcessExitedEventArgs` interfaces lets
+app developers get the `BrowserProcessExitKind` so they can decide how to handle
+different exit kinds or bypass handling if an event handler for the
+`CoreWebView2`s `ProcessFailed` event (for
```

Comment:
Is the order the two events fire guaranteed? Would be good to include that

Reply:
It's not. I added a note about that.